File: scripts/check_footballplayers_table.py
"""
Check footballplayers table structure and data
"""

import os
import psycopg2
from dotenv import load_dotenv

# Load environment variables
load_dotenv('.env.local')

DATABASE_URL = os.getenv('DATABASE_URL') or os.getenv('NEON_DATABASE_URL')

def check_table():
    """Check footballplayers table"""
    
    conn = None
    try:
        print('🔍 Connecting to database...\n')
        conn = psycopg2.connect(DATABASE_URL)
        cur = conn.cursor()
        
        # Get table columns
        print('📊 Checking footballplayers table columns...\n')
        cur.execute("""
            SELECT column_name, data_type, is_nullable
            FROM information_schema.columns
            WHERE table_name = 'footballplayers'
            ORDER BY ordinal_position
        """)
        
        columns = cur.fetchall()
        print(f'Found {len(columns)} columns:')
        for col_name, data_type, nullable in columns:
            print(f'  - {col_name}: {data_type} {"NULL" if nullable == "YES" else "NOT NULL"}')
        
        # Get sample data for player 110599
        print('\n\n📊 Sample data for player_id=110599:\n')
        cur.execute("""
            SELECT *
            FROM footballplayers
            WHERE player_id = %s
        """, ('110599',))
        
        result = cur.fetchone()
        if result:
            colnames = [desc[0] for desc in cur.description]
            for i, col in enumerate(colnames):
                print(f'  {col}: {result[i]}')
        else:
            print('Player not found')
        
        cur.close()
        
    except Exception as error:
        print(f'❌ Error: {error}')
        raise error
    finally:
        if conn:
            conn.close()

if __name__ == '__main__':
    try:
        check_table()
        print('\n✅ Check completed')
    except Exception as error:
        print(f'\n❌ Check failed: {error}')
        exit(1)

File: check-footballplayers-table.js
// Check footballplayers table structure
const { neon } = require('@neondatabase/serverless');
require('dotenv').config({ path: '.env.local' });

const sql = neon(process.env.DATABASE_URL || process.env.NEON_DATABASE_URL);

async function checkTable() {
  try {
    console.log('Checking footballplayers table structure...\n');
    
    // Get all columns
    const columns = await sql`
      SELECT 
        column_name, 
        data_type,
        is_nullable,
        column_default
      FROM information_schema.columns 
      WHERE table_name = 'footballplayers'
      ORDER BY ordinal_position
    `;
    
    console.log('Columns in footballplayers table:\n');
    columns.forEach(col => {
      console.log(`- ${col.column_name}`);
      console.log(`  Type: ${col.data_type}`);
      console.log(`  Nullable: ${col.is_nullable}`);
      if (col.column_default) {
        console.log(`  Default: ${col.column_default}`);
      }
      console.log('');
    });
    
    console.log(`Total columns: ${columns.length}\n`);
    
    // Get a sample record
    console.log('Sample record:\n');
    const sample = await sql`
      SELECT * FROM footballplayers LIMIT 1
    `;
    
    if (sample.length > 0) {
      console.log('Fields in actual data:');
      Object.keys(sample[0]).forEach(key => {
        const value = sample[0][key];
        const valueStr = value === null ? 'NULL' : (typeof value === 'string' && value.length > 50 ? value.substring(0, 50) + '...' : value);
        console.log(`  ${key}: ${valueStr}`);
      });
    }
    
  } catch (error) {
    console.error('Error:', error);
    process.exit(1);
  }
  
  process.exit(0);
}

checkTable();
